runtime/compiler/optimizer/UnsafeFastPath.cpp

```diff
@@ -132,6 +132,19 @@ static bool isVarHandleOperationMethodOnArray(TR::RecognizedMethod rm)
 void TR_UnsafeFastPath::transformUnsafeAtomicCallInVarHandleAccessMethod(TR::TreeTop* callTree, TR::RecognizedMethod callerMethod, TR::RecognizedMethod calleeMethod)
    {
    TR::Node* node = callTree->getNode()->getFirstChild();
+
+   // Give up on arraylet
+   //
+   if (isVarHandleOperationMethodOnArray(callerMethod)
+       && TR::Compiler->om.canGenerateArraylets())
+      {
+      if (trace())
+         {
+         traceMsg(comp(), "Call %p n%dn is accessing an element from an array that might be arraylet, quit\n", node, node->getGlobalIndex());
+         }
+      return;
```

Comment:
This seems a little odd to me. The function documentation and the function name seems to imply the transformation will always be made, yet we bail out here. Doesn't this code belong to something like `isTransformableUnsafeAtomic`? Otherwise I'd like to see the documentation and the name changed to reflect what is actually happening.

Reply:
Good point. `isTransformableUnsafeAtomic` was first introduced as a public API in j9method, since it lives in UnsafeFastPath now, will make it more tailored for VarHandle.

Reply:
@fjeremic I've updated the transform API as well as the documention, can you continue the review?